pkg/resource/deploy/deployment_executor.go

```diff
@@ -409,11 +409,17 @@ func (ex *deploymentExecutor) performDeletes(
 				// This can happen if an earlier create failed, thus the resource wouldn't have been added
 				// to the graph.  Since the resource was just tried to be  created it couldn't have any dependencies
 				// that should be deleted either.
-				if seenErrors.Contains(step) || !ex.deployment.depGraph.Contains(step.Res()) {
+				if seenErrors.Contains(step) {
 					continue
 				}
-				deps := ex.deployment.depGraph.TransitiveDependenciesOf(step.Res())
-				erroredDeps = erroredDeps.Union(deps)
+				if ex.deployment.depGraph.Contains(step.Res()) {
+					deps := ex.deployment.depGraph.TransitiveDependenciesOf(step.Res())
+					erroredDeps = erroredDeps.Union(deps)
+				}
+				if ex.deployment.depGraph.Contains(step.Old()) {
+					oldDeps := ex.deployment.depGraph.TransitiveDependenciesOf(step.Old())
+					erroredDeps = erroredDeps.Union(oldDeps)
+				}
```

Comment:
Not sure if it's worth coding this as a loop to convey that it's the same piece of code/we are collecting multiple instances of the same thing? E.g.

```go
for _, r := range []*resource.State{step.Res(), step.Old()} {
  deps := ex.deployment.depGrap.TransitiveDependenciesOf(r)
  erroredDeps = erroredDeps.Union(deps)
}
```

Happy either way -- great catch!

Reply:
yep, that's nicer, done :heavy_check_mark: